Favot/openai-translation-updater
# Implement Code Formatting and Linting Tools

## Description

In order to maintain a clean and consistent codebase, this PR sets up Prettier, ESLint, and Husky. These tools will help enforce coding standards, identify potential bugs, and automate formatting and linting checks.

### Tasks

- [x] **Setup Prettier**: Install and configure Prettier to automatically format code according to specified rules.
- [x] **Update Code with Prettier Rules**: Ran Prettier on the codebase to apply the initial set of formatting rules.
- [x] **Update Prettier Rules**: Reviewed and updated Prettier rules for better alignment with the project's coding standards.
- [x] **Reformat Files**: Ran Prettier again on the codebase to apply the updated rules.
- [x] **Setup ESLint**: Install and configure ESLint to identify and fix potential code issues.
- [x] **Fix ESLint Issues**: Addressed ESLint warnings and errors, including the removal of `any` type to improve type safety.
- [x] **Setup Husky**: Install and configure Husky to automate linting and formatting checks before commits.

### Testing

- Ensure that ESLint and Prettier run as expected and report issues.
- Make a commit to verify that Husky triggers the linting and formatting checks pre-commit.
- Review the codebase to ensure that formatting has been applied consistently.

### Notes

- Developers should now install project dependencies to get the benefits of the new linting and formatting tools.
- It's recommended to set up your editor to run Prettier and ESLint as you save files for real-time feedback.

---

This implementation should help streamline the development process and ensure a higher code quality standard in the project.

---
